modules/metrics/src/main/java/com/spotify/apollo/metrics/semantic/SemanticServiceMetrics.java

```diff
@@ -187,10 +192,27 @@ private CachedMeters metersForEndpoint(String endpoint) {
   private Optional<Timer> requestDurationTimer(MetricId id) {
     return enabledMetrics.test(ENDPOINT_REQUEST_DURATION) ?
            Optional.of(metricRegistry
-                           .timer(id.tagged("what", ENDPOINT_REQUEST_DURATION.tag()))) :
+               .timer(id.tagged("what", ENDPOINT_REQUEST_DURATION.tag()))) :
            Optional.empty();
   }
 
+  /**
+   * Checks for endpoint-duration-goal configuration options and sets up
+   * metrics that track how many requests meet a certain duration threshold goal.
+   *
+   * @param endpoint formatted as METHOD:URI
+   */
+  private Optional<DurationThresholdTracker> requestDurationThresholdTracker(MetricId id,
+                                                                             String endpoint) {
+    final Optional<Integer> endpointThreshold =
```

Comment:
This can be written as:
```
    return durationThresholdConfig.getDurationThresholdForEndpoint(endpoint)
        .map(threshold -> Optional.of(new DurationThresholdTracker(id, metricRegistry, threshold)))
        .orElse(Optional.empty());
```